homeassistant/components/binary_sensor/__init__.py

```diff
@@ -159,3 +161,15 @@ def state(self):
     def device_class(self):
         """Return the class of this device, from component DEVICE_CLASSES."""
         return None
+
+
+class BinarySensorDevice(BinarySensorEntity):
+    """Represent a binary sensor (for backwards compatibility)."""
+
+    def __init__(self, *args, **kwargs):
```

Comment:
The developer need to call super to have this run. But pylint will force them to do that to satisfy the linter.

Reply:
Home Assistant has this linter set up. However, custom components might not. I suggest to use this instead (from `config_entries.py#ConfigFlow`):

```python
    def __init_subclass__(cls, **kwargs: Any) -> None:
        """Initialize a subclass."""
        _LOGGER.warning("…")
        super().__init_subclass__(**kwargs)  # type: ignore

```

Reply:
Here are the docs:
https://docs.python.org/3/reference/datamodel.html#customizing-class-creation

Reply:
Brilliant!